### PR TITLE
feat: compact mode

### PR DESCRIPTION
**Pull request by nicoglennon**
compacc mode 👛

https://user-images.githubusercontent.com/16931094/183975907-cc8aa339-31db-4b64-add6-5ba09f92c619.mov

### CODE REVIEW

File: packages/rainbowkit/src/components/ConnectOptions/DesktopOptions.tsx [hunk 1]
```diff
@@ -1,28 +1,38 @@
...
-import { ScrollClassName, sidebar } from './DesktopOptions.css';
+import {
+  ScrollClassName,
+  sidebar,
+  sidebarCompactMode,
+} from './DesktopOptions.css';
 
 export enum WalletStep {
   None = 'NONE',
+  ProLearn = 'PRO_LEARN',
```
Comment (markdalgleish) [inline packages/rainbowkit/src/components/ConnectOptions/DesktopOptions.tsx:35]: Should be `Compact` now.

Also, tiny nit: I feels like it reads better as `LearnCompact` / `LEARN_COMPACT`.

Comment (nicoglennon) [reply packages/rainbowkit/src/components/ConnectOptions/DesktopOptions.tsx:35]: agreed. missed this one on the search

File: packages/rainbowkit/src/components/ConnectOptions/DesktopOptions.tsx [hunk 2]
```diff
@@ -174,61 +209,149 @@ export function DesktopOptions({ onClose }: { onClose: () => void }) {
           wallet={selectedWallet}
         />
       );
-      headerLabel = hasQrCode && `Get started with ${selectedWallet.name}`;
+      headerLabel =
```
Comment (markdalgleish) [inline packages/rainbowkit/src/components/ConnectOptions/DesktopOptions.tsx:212]: Just confirming, we want to use the short name even when we're not in compact mode?

File: packages/rainbowkit/src/components/ConnectOptions/DesktopOptions.tsx [hunk 3]
```diff
@@ -174,61 +209,149 @@ export function DesktopOptions({ onClose }: { onClose: () => void }) {
...
+        hasQrCode &&
+        `Get started with ${selectedWallet.shortName || selectedWallet.name}`;
```
Comment (markdalgleish) [inline packages/rainbowkit/src/components/ConnectOptions/DesktopOptions.tsx:214]: Just confirming, we want to use the short name even when we're not in compact mode?

Comment (nicoglennon) [reply packages/rainbowkit/src/components/ConnectOptions/DesktopOptions.tsx:214]: yeah - this really is just for coinbase, "coinbase wallet" felt crowded in this screen even on default mode. i can make it the long string on default to keep consistent as much as possible 👍

**Comment by markdalgleish** [comment]
I'm wondering whether we need to avoid using a boolean prop so we don't corner ourselves if we need to add additional modes in the future, e.g. `<RainbowKitProvider modalSize="compact">`.

(I know we already have `coolMode`, but that one's deliberately tongue in cheek 😄)

File: .changeset/nice-rockets-pay.md [hunk 4]
```diff
@@ -0,0 +1,5 @@
+---
+'@rainbow-me/rainbowkit': patch
+---
+
+Add a 'Compact' modal size option for developers looking to use a simpler version of Rainbowkit
```
Comment (markdalgleish) [inline .changeset/nice-rockets-pay.md:5]:
```suggestion
Add a 'compact' modal size option for developers looking to use a simpler version of RainbowKit, available by setting the `modalSize` prop to `"compact"` on `RainbowKitProvider`.
```

**Review by markdalgleish — APPROVED** [review]
Just a small note about adding some more detail to the changeset, but otherwise this looks great 🔥🔥

**Pull request merged**